extensions/mssql/src/ui/dialogBase.ts

```diff
@@ -164,13 +164,21 @@ export abstract class DialogBase<DialogResult> {
 		return inputBoxComponent;
 	}
 
-	protected createInputBox(ariaLabel: string, textChangeHandler: (newValue: string) => Promise<void>, value: string = '', enabled: boolean = true, type: azdata.InputBoxInputType = 'text', width: number = DefaultInputWidth, required?: boolean, min?: number, max?: number): azdata.InputBoxComponent {
+	protected createInputBox(ariaLabel: string, textChangeHandler: (newValue: string) => Promise<void>, value: string = '', enabled: boolean = true, type: azdata.InputBoxInputType = 'text', width: number = DefaultInputWidth, required?: boolean, min?: number, max?: number, enterKeyPropagationOnly: boolean = false): azdata.InputBoxComponent {
```

Comment:
Maybe `validationOnEnterKey` instead? Although shouldn't the input box be handling this already for the text changing?

And why do you only want to validate on enter anyways? Is the validation doing something besides the normal text validation that you don't want firing off a lot?

Reply:
Reason for doing this is, custom dialog.ts createInputBox only triggering on text change and nothing on enterKeyPress. For the DSC Input types, since we are updating the table on input value change, textchange is causing delay due to the refresh and losing focus in input box. 

So allowing enter key press only to update the table and don't want to fire the onTextChange handler in this case.

Reply:
That was still an issue with this change, but able to fix the input type focus and table focus with the latest commit, by reverting the EnterKeyPropagation change, and added a video to the description. With this approach we are able to refresh the table data with input value and still putting focus on the user input box (primary and secondary, which ever user is working on).